pkgs/desktops/gnome-3/extensions/update-extensions.py

```diff
@@ -156,51 +160,71 @@ def process_extension(extension: Dict[str, Any]) -> Optional[Dict[str, Any]]:
         "pname": pname,
         "description": sanitize_string(extension["description"]),
         "link": "https://extensions.gnome.org" + extension["link"],
-        "shell_version_map": shell_version_map
+        "shell_version_map": shell_version_map,
     }
 
 
 def fetch_extensions() -> List[Dict[str, Any]]:
     """
     Fetch the extensions. The extensions database is scraped by using the search query. We simply go over it
     page by page until we hit a non-full page or a 404 error.
+
+    The returned list is sorted by the age of the extension, in order to be deterministic.
     """
     page = 0
     extensions = []
     while True:
         page += 1
-        logging.info("")
         logging.info("Scraping page " + str(page))
         try:
             with urllib.request.urlopen(
                 f"https://extensions.gnome.org/extension-query/?n_per_page=25&page={page}"
             ) as url:
                 data = json.loads(url.read().decode())["extensions"]
                 responseLength = len(data)
-                logging.debug(f"Found {responseLength} extensions")
 
                 for extension in data:
-                    extension = process_extension(extension)
-                    if extension:
-                        extensions += [extension]
+                    extensions += [extension]
 
                 # If our page isn't "full", it must have been the lasts one
                 if responseLength < 25:
+                    logging.debug(
+                        f"\tThis page only has {responseLength} entries, so it must be the last one."
+                    )
                     break
         except urllib.error.HTTPError:
             # Assume this error is a 404. We hit the last page; we're done.
             break
+
+    from operator import itemgetter
```

Comment:
No, just move this to the top of the file.

Reply:
It's really interesting how the same coding best practices for one language can be bad style or even anti-patterns in another.